src/frontend/src/components/common/Header/Header.tsx

```diff
@@ -1,12 +1,101 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import BCGovLogo from '../../../assets/BC_Logo_Horizontal.svg';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faUser } from '@fortawesome/free-solid-svg-icons';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { API_ROUTES } from '../../../constant/apiRoutes';
+import { routes } from '../../../constant/routes';
+import Modal from '../Modal';
+import { HttpRequest } from '../../../utils/http-request.util';
 
 type Props = {
   user: string | null;
 };
 function Header({ user }: Props) {
+  const navigate = useNavigate();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [accessToken, setAccessToken] = useState(
+    localStorage.getItem('access_token'),
+  );
+  const [showModal, setShowModal] = useState(false);
+  const [userInfo, setUserInfo] = useState(null);
+  const [message, setMessage] = useState('');
+  const code = searchParams.get('code');
+  // https://github.com/microsoft/TypeScript/issues/48949
+  // workaround
+  const win: Window = window;
+
+  useEffect(() => {
```

Comment:
I would pull this auth code out into either a custom wrapper component that adds context, or a custom hook. Mixing it in with the `Header` code is harder to read.

Reply:
I create a custom hook. please review

Reply:
@will-freshworks can you review bing's changes please 😄